backend/corpora/lambdas/api/v1/collection_uuid/publish.py

```diff
@@ -1,14 +1,38 @@
 from flask import make_response, g
 
 from .....common.corpora_orm import CollectionVisibility
-from .....common.entities import Collection
+from .....common.entities import Collection, DatasetAsset
 from .....common.utils.exceptions import ConflictException
 
 from .....api_server.db import dbconnect
-from .....common.utils.exceptions import ForbiddenHTTPException
+from .....common.utils.exceptions import ForbiddenHTTPException, ServerErrorHTTPException
 from backend.corpora.lambdas.api.v1.collection import _owner_or_allowed
 
 
+def check_for_duplicate_datasets(collection: Collection) -> bool:
+    """
+    Check if duplicate datasets are found. Return true on the first duplicate match.
+    :param collection:
+    :return: True if duplicates detected, otherwise return false.
+    """
+    etags = []
```

Comment:
intentional micro-optimization here by using list instead of set? For small data structure (since we're talking < 100 items)?